src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/KestrelThread.cs

```diff
@@ -160,6 +160,11 @@ public Task StartAsync()
                 if (!await ConnectionManager.WalkConnectionsAndCloseAsync(_shutdownTimeout).ConfigureAwait(false))
                 {
                     _log.NotAllConnectionsClosedGracefully();
+
+                    if (!await ConnectionManager.WalkConnectionsAndAbortAsync(_shutdownTimeout).ConfigureAwait(false))
```

Comment:
I don't think I like repeating the `_shutdownTimeout` here. That could be configured to a very large value. This probably doesn't need to be configurable, since this isn't really a timeout meant for running application code.

I would probably make it 1 second.